components/LocationList.js

```diff
@@ -1,19 +1,26 @@
-import Card from "./Card";
+import LocationCard from "./LocationCard";
 import styled from "styled-components";
+import Link from "next/link";
+import { StyledList } from "./StyledList";
 
-const StyledList = styled.ul`
-  display: flex;
-  flex-direction: column;
-  align-items: center;
-  padding-left: 0;
+const StyledLink = styled(Link)`
+  text-decoration: none;
 `;
 
 export default function LocationList({ locations }) {
   return (
     <StyledList>
-      {locations.map((location, index) => (
-        <Card location={location} key={index} />
-      ))}
+      {locations.map((location, id) => {
+        return (
+          <StyledLink
+            href={`/locationdetail/${id}`}
+            key={id}
```

Comment:
you are still using the index as a key. (The second parameter of the map function is always the index). I recommend the location object should have an id. For this purpose you should use the handleAddLocation function to give an id to each new created location. Here you can use 'nanoid()' to give a unique id. If you have questions to this point leave us a message.

Reply:
Hello @Roland-Hufnagel, I tried using nanoid() but couldn't figure out how. When I adjust the handleAddLocation function to

` setLocations([...locations, { id: nanoid(), location: newLocation }]);` 

I get the following error message:

`Error: Objects are not valid as a React child (found: object with keys {location}). If you meant to render a collection of children, use an array instead.`

How can I solve that? 

Moreover, I also tried to leave the handleAddLocation function the same but adding 

`const id = nanoid()`

in the LocationList component before the return statement. Then, a nanoid is assigned but when clicking on a location and being redirected on the location's detail page, no location card is shown (I want to show the location card that was clicked on as a heading for the bee colony list here). 

Thanks a lot for your help!

Reply:
I think you should use it like that:
setLocations([...locations, {...newLocation, id: nanoid()}]);

Reply:
Thank you, but it doesn't seem to work. It renders everything correctly and there is no error message, but the IDs of the cards are 0, 1 and so on. It seems as if nanoid is not applied?